src/ds/json_schema.h

```diff
@@ -83,18 +200,28 @@ namespace ds
         nonstd::is_specialization<T, std::map>::value ||
         nonstd::is_specialization<T, std::unordered_map>::value)
       {
-        // Nlohmann serialises maps to an array of (K, V) pairs
-        schema["type"] = "array";
-        auto items = nlohmann::json::object();
+        // Nlohmann serialises maps to an array of (K, V) pairs...
+        if (std::is_same<typename T::key_type, std::string>::value)
         {
-          items["type"] = "array";
+          // ...unless the keys are strings!
```

Comment:
That's because json only allows strings as keys in maps:

![image](https://user-images.githubusercontent.com/4016369/93571830-e6bb6580-f98c-11ea-83dd-423168de14ed.png)

Reply:
Yes - I understand why nlohmann does this, just trying to make sure the code explains these branches as its not obvious they're describing an nlohmann implementation detail (it _could_ have decided that string-keyed maps were also pair-lists, not objects).